documentation/authentication.md

```diff
@@ -0,0 +1,139 @@
+
+> **NOTE:** API Key Authentication should only be used when [TLS is enabled](#) to protect the key while in transit.
+
+### Generating an API Key
+
+The API Key you use to secure `dotnet-monitor` should be a 32-byte cryptographically random secret. You can generate a key either using `dotnet-monitor` or via your shell. To generate an API Key with `dotnet-monitor`, simply invoke the `generatekey` command:
+
+```powershell
+dotnet-monitor generatekey
+```
+
+The output from this command will display the API Key formatted as an authentication header along with its hash and associated hashing algorithm. You will need to store the `ApiKeyHash` and `ApiKeyHashType` in the configuration for `dotnet-monitor` and use the authorication header value when making requests to the `dotnet-monitor` HTTPS endpoint.
+
+```
+Authorization: MonitorApiKey Uf6Yq8ZGkcn+ltq2QLHxuXpKA6/Q5VH5Mb5aSIJBxRc=
+ApiKeyHash: CB233C3BE9F650146CFCA81D7AA608E3A3865D7313016DFA02DAF82A2505C683
+ApiKeyHashType: SHA256
+```
+
+> NOTE: The generated API Key should be secured at rest. We recommend using a tool such as a password manager to save it.
+
+### Configuring dotnet-monitor to use an API Key
+
+Using the `generatekey` command does not automatically update the configuration of `dotnet-monitor` to use the new key. You can update the configuration of `dotnet-monitor` via settings file, environemt variable of kubernetes secrets.
+
+If you're running on Windows, you can save these settings to `%USERPROFILE%\.dotnet-monitor\settings.json`. If you're on other operating systems, you can save this file to `$XDG_CONFIG_HOME/dotnet-monitor/settings.json`.
+
+> **NOTE:** If `$XDG_CONFIG_HOME` isn't defined, dotnet-monitor will fall back to looking at `$HOME/.config/dotnet-monitor/settings.json`
+
+```json
+{
+  "ApiAuthentication": {
+    "ApiKeyHash": "CB233C3BE9F650146CFCA81D7AA608E3A3865D7313016DFA02DAF82A2505C683",
+    "ApiKeyHashType": "SHA256"
+  }
+}
+```
+
+Alternatively, you can use environment variables to specify the configuration.
+
+```sh
+DotnetMonitor_ApiAuthentication__ApiKeyHash="CB233C3BE9F650146CFCA81D7AA608E3A3865D7313016DFA02DAF82A2505C683"
+DotnetMonitor_ApiAuthentication__ApiKeyHashType="SHA256"
+```
+
+> **NOTE:** When you use environment variables to configure the API Key hash, you must restart `dotnet-monitor` for the changes to take effect.
+
+### Configuring an API Key in a Kubernetes Cluster
+If you're running in Kubernetes, we recommend creating secrets and mounting them into container via a deployment manifest. You can use this `kubectl` command to either create or rotate the API Key.
+
+```sh
+kubectl create secret generic apikey \
+	--from-literal=ApiAuthentication__ApiKeyHash=$hash \
+	--dry-run=client -o yaml \
```

Comment:
I would add the --from-literal=ApiAuthentication__ApiKeyHashType=SHA256 line here too. K8 will mount both settings into the file system.

Reply:
Related to this is https://github.com/dotnet/dotnet-monitor/issues/164. We don't make it easy to mount secret and non-secret data from multiple sources.